pkg/istio/rbac/v2/provider.go

```diff
@@ -0,0 +1,136 @@
+package v2
+
+import (
+	"github.com/yahoo/athenz/clients/go/zms"
+	"github.com/yahoo/k8s-athenz-istio-auth/pkg/athenz"
+	m "github.com/yahoo/k8s-athenz-istio-auth/pkg/athenz"
+	"github.com/yahoo/k8s-athenz-istio-auth/pkg/istio/common"
+	"github.com/yahoo/k8s-athenz-istio-auth/pkg/istio/rbac"
+	"github.com/yahoo/k8s-athenz-istio-auth/pkg/log"
+	"istio.io/api/security/v1beta1"
+	workloadv1beta1 "istio.io/api/type/v1beta1"
+	"istio.io/istio/pilot/pkg/model"
+	"istio.io/istio/pkg/config/schema/collections"
+	"regexp"
+)
+
+// implements github.com/yahoo/k8s-athenz-istio-auth/pkg/istio/rbac/Provider interface
+type v2 struct {enableOriginJwtSubject bool}
+
+func NewProvider(enableOriginJwtSubject bool) rbac.ProviderV2 {
+	return &v2{
+		enableOriginJwtSubject: enableOriginJwtSubject,
+	}
+}
+
+func (p *v2) ConvertAthenzModelIntoIstioAuthzPolicy(athenzModel athenz.Model, namespace string, serviceName string, svcLabel string) model.Config {
+	// authz policy is created per service. each rule is created by each role, and form the rules under
+	// this authz policy.
+	var out model.Config
+	// form authorization config meta
+	// namespace: service's namespace
+	// name: service's name
+	schema := collections.IstioSecurityV1Beta1Authorizationpolicies
+	out.ConfigMeta = model.ConfigMeta{
+		Type:      schema.Resource().Kind(),
+		Group:     schema.Resource().Group(),
+		Version:   schema.Resource().Version(),
+		Namespace: namespace,
+		Name:      serviceName,
+	}
+
+	// matching label, same with the service label
+	spec := &v1beta1.AuthorizationPolicy{}
+
+	spec.Selector = &workloadv1beta1.WorkloadSelector{
+		MatchLabels: map[string]string{"svc": svcLabel},
+	}
+
+	// generating rules, iterate through assertions, find the one match with desired format.
+	var rules []*v1beta1.Rule
+	for role, assertions := range athenzModel.Rules {
+		for _, assert := range assertions {
+			// assert.Resource contains the svc information that needs to parse and match
+			svc, path, err := common.ParseAssertionResource(zms.DomainName(m.NamespaceToDomain(namespace)), assert)
+			if err != nil {
+				continue
+			}
+			// if svc match with current svc, process it and add it to the rules
+			// note that svc defined on athenz can be a regex, need to match the pattern
+			res, e := regexp.MatchString(svc, svcLabel)
+			if e != nil {
+				log.Errorln("error matching string: ", e.Error())
+				continue
+			}
+			if res {
```

Comment:
Should we check for !res here and continue directly? This way we can avoid indenting the subsequent logic.